url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParserTokenManager.java

```diff
@@ -0,0 +1,1654 @@
+            return jjStartNfaWithStates_0(3, 20, 48);
+         else if ((active0 & 0x4000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 38, 48);
+         else if ((active0 & 0x80000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 43, 48);
+         return jjMoveStringLiteralDfa4_0(active0, 0x810000e00200000L, active1, 0L);
+      case 72:
+      case 104:
+         if ((active1 & 0x200L) != 0L)
+            return jjStartNfaWithStates_0(3, 73, 48);
+         break;
+      case 73:
+      case 105:
+         return jjMoveStringLiteralDfa4_0(active0, 0x80000000L, active1, 0L);
+      case 75:
+      case 107:
+         if ((active0 & 0x8000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 39, 48);
+         else if ((active1 & 0x400L) != 0L)
+            return jjStartNfaWithStates_0(3, 74, 48);
+         break;
+      case 76:
+      case 108:
+         if ((active0 & 0x800000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 47, 48);
+         else if ((active0 & 0x100000000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 56, 48);
+         return jjMoveStringLiteralDfa4_0(active0, 0x4200000001002000L, active1, 0x2L);
+      case 77:
+      case 109:
+         if ((active0 & 0x20000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 29, 48);
+         return jjMoveStringLiteralDfa4_0(active0, 0x30000L, active1, 0L);
+      case 78:
+      case 110:
+         return jjMoveStringLiteralDfa4_0(active0, 0xc0000L, active1, 0x4L);
+      case 79:
+      case 111:
+         if ((active0 & 0x1000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 36, 48);
+         return jjMoveStringLiteralDfa4_0(active0, 0x20000000000000L, active1, 0x8L);
+      case 82:
+      case 114:
+         if ((active0 & 0x8000L) != 0L)
+            return jjStartNfaWithStates_0(3, 15, 48);
+         return jjMoveStringLiteralDfa4_0(active0, 0xa000000000000000L, active1, 0x100L);
+      case 83:
+      case 115:
+         return jjMoveStringLiteralDfa4_0(active0, 0x2000000L, active1, 0L);
+      case 84:
+      case 116:
+         if ((active0 & 0x4000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 26, 48);
+         else if ((active0 & 0x40000000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 54, 48);
+         return jjMoveStringLiteralDfa4_0(active0, 0x800000L, active1, 0x800L);
+      case 85:
+      case 117:
+         return jjMoveStringLiteralDfa4_0(active0, 0x140040000000L, active1, 0x20L);
+      case 87:
+      case 119:
+         return jjMoveStringLiteralDfa4_0(active0, 0x800L, active1, 0L);
+      case 89:
+      case 121:
+         if ((active0 & 0x4000000000000L) != 0L)
+            return jjStartNfaWithStates_0(3, 50, 48);
+         break;
+      default :
+         break;
+   }
+   return jjStartNfa_0(2, active0, active1);
+}
+static private int jjMoveStringLiteralDfa4_0(long old0, long active0, long old1, long active1)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 74 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.